src/query/storage/converter_test.go

```diff
@@ -353,28 +361,106 @@ func TestPromTimeSeriesToSeriesAttributesM3Type(t *testing.T) {
 	}
 
 	for proto, expected := range mapping {
-		attrs, err := PromTimeSeriesToSeriesAttributes(prompb.TimeSeries{M3Type: proto})
+		attrs, err := PromTimeSeriesToSeriesAttributes(prompb.TimeSeries{
+			Source: prompb.Source_PROMETHEUS,
+			M3Type: proto,
+		})
 		require.NoError(t, err)
 		assert.Equal(t, expected, attrs.M3Type)
+		assert.Equal(t, ts.SourceTypePrometheus, attrs.Source)
+	}
+
+	_, err := PromTimeSeriesToSeriesAttributes(prompb.TimeSeries{
+		Source: prompb.Source_PROMETHEUS,
+		M3Type: -1,
+	})
+	require.Error(t, err)
+}
+
+func TestPromTimeSeriesToSeriesAttributesMetricsTypeFromOpenMetrics(t *testing.T) {
+	mapping := map[prompbMetricTypeWithNameSuffix]promMetricTypeWithBool{
+		{metricType: prompb.MetricType_UNKNOWN}: {metricType: ts.PromMetricTypeUnknown},
+
+		{prompb.MetricType_COUNTER, "total"}:   {metricType: ts.PromMetricTypeCounter, handleValueResets: true},
+		{prompb.MetricType_COUNTER, "created"}: {metricType: ts.PromMetricTypeCounter},
+
+		{metricType: prompb.MetricType_GAUGE}:    {metricType: ts.PromMetricTypeGauge},
+		{metricType: prompb.MetricType_INFO}:     {metricType: ts.PromMetricTypeInfo},
```

Comment:
nit: According to spec, metric type info should have `_info` suffix. Though not sure if it's worth adding, since it doesn't affect the result.

Reply:
Yeah, I decided to be flexible here, as there is no need to differentiate metric type members.